src/QueryReflection/QueryReflection.php

```diff
@@ -104,11 +111,18 @@ public function getResultType(string $queryString, int $fetchType): ?Type
         }
 
         $resultType = self::reflector()->getResultType($queryString, $fetchType);
-        if (
-            null !== $resultType
-            && self::getRuntimeConfiguration()->isStringifyTypes()
-        ) {
-            return $this->stringifyResult($resultType);
+
+        if (null !== $resultType) {
+            if (!$resultType instanceof ConstantArrayType) {
+                throw new ShouldNotHappenException();
+            }
+
+            $parserInference = new ParserInference($this->getSchemaReflection());
+            $resultType = $parserInference->narrowResultType($queryString, $resultType);
```

Comment:
we should make this inference disabled by default and allow to enable it via `RuntimeConfiguration` for now.

reasons
- stabilize
- sqlftw atm requires php 7.4, so this feature does not work for all versions supported by phpstan-dba